nemo/collections/asr/parts/collections.py

```diff
@@ -335,3 +335,123 @@ def __parse_item(self, line: str, manifest_file: str) -> Dict[str, Any]:
         )
 
         return item
+
+
+class FeatSeqLabel(_Collection):
+    """List of feature sequence of label correspondence with preprocessing."""
+
+    OUTPUT_TYPE = collections.namedtuple(typename='FeatSeqLabelEntity', field_names='feature_file seq_label',)
+
+    def __init__(
+        self,
+        feature_files: List[str],
+        seq_labels: List[str],
+        max_number: Optional[int] = None,
+        index_by_file_id: bool = False,
+    ):
+        """Instantiates feat-SeqLabel manifest with filters and preprocessing.
+        """
+        output_type = self.OUTPUT_TYPE
+        data, num_filtered = (
+            [],
+            0.0,
+        )
+        self.uniq_labels = set()
+
+        if index_by_file_id:
+            self.mapping = {}
+
+        for feature_file, seq_label in zip(feature_files, seq_labels):
+
+            label_tokens, uniq_labels_in_seq = self.dnc_parser(seq_label)
+
+            data.append(output_type(feature_file, label_tokens))
+            self.uniq_labels |= uniq_labels_in_seq
+
+            if label_tokens is None:
+                num_filtered += 1
+                continue
+
+            if index_by_file_id:
+                file_id, _ = os.path.splitext(os.path.basename(feature_file))
+                self.mapping[feature_file] = len(data) - 1
+
+            # Max number of entities filter.
+            if len(data) == max_number:
+                brak
+
+        logging.info("# {} files loaded including # {} unique labels".format(len(data), len(self.uniq_labels)))
+        super().__init__(data)
+
+    def dnc_parser(self, seq_label):
```

Comment:
Define what is dnc here or change name of function ?

Reply:
yes I should eliminate ambiguous dnc here.